### PR TITLE
Update packets up to 2018-06-27 and fix some existing packets

### PR DESCRIPTION
**Pull request by 4144**
### Pull Request Prelude

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR will be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

Update packets up to 2018-06-27.
Fix packet 836 (search in stores).
Added item options fields into this packet.
Updated other things for support for report item options in vending.
Buying store not supported for now.
Add packets for warp and change map server on airship.
Add new packet for pincode login state.
Add new auth error packet in zero.
Fix roulette for 2018 clients.

### CODE REVIEW

**Comment by HerculesWSAPI** [comment]
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/2095)

File: src/char/pincode.c [hunk 1]
```diff
@@ -249,6 +253,32 @@ void pincode_loginstate(int fd, struct char_session_data* sd, enum pincode_login
...
+// 2 = create new pin - client sends 0x8ba
+// 3 = pin must be changed - client 0x8be
+// 4 = create new pin ?? - client sends 0x8ba
+// 5 = client shows msgstr(1896)
+// 6 = client shows msgstr(1897) Unable to use your KSSN number
+// 7 = char select window shows a button - client sends 0x8c5
+// 8 = pincode was incorrect
+// [4144] pincode_loginstate2 can replace pincode_loginstate,
+// but kro using pincode_loginstate2 only for send wrong pin error or locked after 3 pins wrong
+void pincode_loginstate2(int fd, struct char_session_data* sd, enum pincode_login_response state, enum pincode_login_response2 flag)
```
Comment (sctnightcore) [inline src/char/pincode.c]: Packet is CZ or ZC ?

Comment (4144) [reply src/char/pincode.c]: not CZ and not ZC. this is char server...

Comment (dastgirp) [reply src/char/pincode.c]: isn't it ZC? because it sends from server to client

Comment (sctnightcore) [reply src/char/pincode.c]: Maybe PACKET_HC_SECOND_PASSWD_LOGIN2 ?

Comment (4144) [reply src/char/pincode.c]: yes this HC packet. it totally same with old one. But for some reason kro using new packet only for message with wrong pin and blocked pin for 3 days (hercules cant do this for now)

**Pull request merged**